File: 프로그래머스/0/120894. 영어가 싫어요/영어가 싫어요.cpp
#include <string>
#include <vector>

using namespace std;

long long solution(string numbers) {
    long long answer = 0;
    vector<string> num = {
        "zero", "one", "two", "three", "four", "five",
        "six", "seven", "eight", "nine"
    };
    int tmp;
    
    for (int i=0; i<num.size(); i++)
    {
        while ((tmp = numbers.find(num[i])) != string::npos)
        {
            numbers.replace(tmp, num[i].size(), to_string(i));
        }
    }
    answer = stoll(numbers);
    return answer;
}

File: 프로그래머스/0/120894. 영어가 싫어요/영어가 싫어요.cs
using System;
using System.Collections.Generic;

public class Solution {
    public long solution(string numbers) {
        List<string> lst = new List<string>()
        {
            "zero", "one", "two", "three", "four",
            "five", "six", "seven", "eight", "nine"
        };

        for (int i = 0; i < lst.Count; i++)
        {
            numbers = numbers.Replace(lst[i], i.ToString());
        }

        return long.Parse(numbers);
    }
}